craft_platforms/_build.py

```diff
@@ -54,6 +55,10 @@ def get_build_plan(
         "build_base": project_data.get("build-base"),
     }
 
+    # Bare bases require a build_base
+    if args["base"] == "bare" and args["build_base"] is None:
+        raise _errors.NeedBuildBaseError(base=args["base"])
```

Comment:
sorry for getting to this late but this is not necessarily true for every application that uses the default build planner; some apps don't support base `bare` at all at the moment.
@lengau is there a strong argument against keeping this check rock-specific?

Reply:
> If an application doesn't accept bare, it should never hit this anyway as the app should be erroring before it tries to create the build plan.

Yes, but Launchpad also uses this code doesn't it? My concern is that someone trying to use 'base: bare' on a profile file that doesn't support `bare` will get something like "base 'bare' needs a build-base", which is wrong if the application doesn't support that base at all.

> On the flip side, it is strictly incorrect to not be able to provide a build base, and bare is a special case of a base that cannot be a build base.

It is a special case indeed, in the sense that some applications that use the default build planner don't support 'bare'.

Reply:
Sorry to insist on this, but now I'm a bit confused. I took the `test_global_environment()` integration test in craft-application and debugged it:
- With `--destructive-mode`, the project is loaded before the build plan so as you say the application would catch the wrong base
- However, if I drop `--destructive-mode` to get a regular run, the build plan is created first (the call is on `_run_manager_for_build_plan()`, which means that this code would be hit before model validation

Reply:
[From offline conversation: we're going to move this back to the rockcraft planner and figure out whether how craft-platforms should handle bare base]